src/v/kafka/server/handlers/fetch.cc

```diff
@@ -893,6 +1387,32 @@ kafka::fetch_plan make_simple_fetch_plan(op_context& octx) {
 }
 } // namespace testing
 
+namespace {
+ss::future<> do_fetch(op_context& octx) {
+    switch (config::shard_local_cfg().fetch_read_strategy) {
+    case model::fetch_read_strategy::polling: {
+        // first fetch, do not wait
+        co_await fetch_topic_partitions(octx).then([&octx] {
+            return ss::do_until(
+              [&octx] { return octx.should_stop_fetch(); },
+              [&octx] { return fetch_topic_partitions(octx); });
+        });
+    } break;
+    case model::fetch_read_strategy::non_polling: {
+        auto planner = make_fetch_planner<simple_fetch_planner>();
+        auto fetch_plan = planner.create_plan(octx);
+
+        fetch_plan_executor executor
+          = make_fetch_plan_executor<nonpolling_fetch_plan_executor>();
```

Comment:
you can just delete this `make_fetch_plan_executor` thing? It is never called polymophically. So you can just have the two cases as here and call functions directly.

Reply:
Right you are, removing the unneeded alloc.